### ISSUE TITLE
[2.5.2-dev]  Frequent Gateway errors and disconnects

### ISSUE THREAD

**Issue by MarcG2** [get-got/discord-downloader-go#137]
See screenshots below.  These disconnects have occurred 4 times within a 10 hour period.  Is this a server side issue?  

I recently started running this app on a different system with a global VPN running.  The downloader is set to be fully excluded from the VPN since I found it was unable connect with it running.  But I wonder if it's still somehow causing issues.

![Screenshot 2024-02-14 164832](https://github.com/get-got/discord-downloader-go/assets/44528943/9a8dd1a7-9c6f-4dd1-9058-22d195daf42c)

**Comment by kauffy** [comment]
This looks like an error on your side having to do with your network configuration. Are you using randomized MAC id on your machine or router or anything?

**Comment by MarcG2** [comment]
> Are you using randomized MAC id on your machine or router or anything?
No, I haven't configured my system to do that.  I don't think my VPN, Private Internet Access, is doing it either.  But I'll leave the VPN off for a while since I've got no better ideas.  

I started using the VPN right before Discord started having 403 errors on version 2.4 of the downloader.  I don't think I was getting these disconnect errors then.

Is there an easy way to have the app log to text file?  When I checked today, the app seems to consistently crash after about a day or two of uptime.

**Comment by MarcG2** [comment]
It turns out my issue was in fact caused by the VPN.  PIA has an option called "advanced kill switch" and apparently having that enabled was causing my disconnect problems.

**Comment by get-got** [comment]
Thanks for the follow up, good to know
